### PR TITLE
Add support for Nimiq Albatross PoS transaction serialization

### PR DESCRIPTION
**Pull request by sisou**
## Description

The transaction format for both signing and sending changed slightly for Nimiq Albatross PoS, to encode optional sender data and support more signature methods (now also supports ECDSA and WebAuthn).

## How to test

To test, connect your wallet to a Nimiq PoS testnet node (provided in our joined Telegram channel). To be valid in the testnet, the `NETWORK_ID` has to be set to `5`.

## Types of changes

* Breaking change: update transaction serialization for signing and sending

## Checklist

- [ ] Create pull request as draft initially, unless its complete.
- [x] Add tests to cover changes as needed.
- [ ] Update documentation as needed.
- [ ] If there is a related Issue, mention it in the description.

If you're adding a new blockchain

- [ ] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain.

### CODE REVIEW

**Comment by PandaKitten96** [comment]
I wanna fuck you so bad

**Comment by PandaKitten96** [comment]
Like fuck you so good and slow

File: src/Nimiq/Transaction.cpp [hunk 1]
```diff
@@ -42,6 +43,7 @@ std::vector<uint8_t> Transaction::getPreImage() const {
     encode32BE(vsh, data);
     data.push_back(NETWORK_ID);
     data.push_back(EMPTY_FLAGS);
+    data.push_back(0x00); // Sender Data size (+ 0 bytes of data)
```
Comment (satoshiotomakan) [inline src/Nimiq/Transaction.cpp:46]: Could you please add a link to the source code somewhere at the beginning of the function?
https://github.com/nimiq/core-rs-albatross/blame/b8ed402c9096ffb54afea52347b91ab7831e75de/primitives/transaction/src/lib.rs#L582

**Review by satoshiotomakan — APPROVED** [review]
I'll add the required changes in a separate PR, thank you!

**Pull request merged**